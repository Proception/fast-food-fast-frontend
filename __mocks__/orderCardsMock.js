import { getOrders, getUserOrders } from '../src/actions/orderActions';
const props = {
    orders: [
        {
          "id": 1,
          "orderNo": "12",
          "price": 5000,
          "orderDate": "30 August, 2016",
          "orderStatus": "Pending",
          "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
        },
        {
          "id": 2,
          "orderNo": "54",
          "price": 1200,
          "orderDate": "Boiled Yam",
          "orderStatus": "In Progress",
          "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
        },
        {
          "id": 3,
          "orderNo": "67",
          "price": 1500,
          "orderDate": "30 October, 2017",
          "orderStatus": "Fulfilled",
          "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
        },
        {
          "id": 4,
          "orderNo": "33",
          "price": 3000,
          "orderDate": "30 January, 2018",
          "orderStatus": "Completed",
          "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
        },
      ],
    order: {
      orders: [{
        "id": 1,
        "orderNo": "12",
        "price": 5000,
        "orderDate": "30 August, 2016",
        "orderStatus": "Pending",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },
      {
        "id": 2,
        "orderNo": "54",
        "price": 1200,
        "orderDate": "Boiled Yam",
        "orderStatus": "In Progress",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },],
      userOrders: [{
        "id": 1,
        "orderNo": "12",
        "price": 5000,
        "orderDate": "30 August, 2016",
        "orderStatus": "Pending",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },
      {
        "id": 2,
        "orderNo": "54",
        "price": 1200,
        "orderDate": "Boiled Yam",
        "orderStatus": "In Progress",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },]
    },
    getOrders,
    getUserOrders
    };
const emptyOrderProps = {
  orders: [
      {
        "id": 1,
        "orderNo": "12",
        "price": 5000,
        "orderDate": "30 August, 2016",
        "orderStatus": "Pending",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },
      {
        "id": 2,
        "orderNo": "54",
        "price": 1200,
        "orderDate": "Boiled Yam",
        "orderStatus": "In Progress",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },
      {
        "id": 3,
        "orderNo": "67",
        "price": 1500,
        "orderDate": "30 October, 2017",
        "orderStatus": "Fulfilled",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },
      {
        "id": 4,
        "orderNo": "33",
        "price": 3000,
        "orderDate": "30 January, 2018",
        "orderStatus": "Completed",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },
    ],
  order: {
    orders: [],
    userOrders: [],
  },
  getOrders,
  getUserOrders
  };
export { props, emptyOrderProps };