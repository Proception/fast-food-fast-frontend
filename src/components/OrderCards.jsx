import React, { Component } from 'react';

const OrderCards = ({ orders }) => (
  <div>
  {
    orders.map(item => (
      <div class="new-order" key={item.id}>
        <div class="order-details">
          <p>
            <label for="orderno">Order No. :</label><span>{item.orderNo}</span>
          </p>
          <p>
            <label for="orderdate">Order Date :</label><span>{item.orderDate}</span>
          </p>
          <p>
            <label for="totalprice">Total Amount :</label><span>â‚¦{item.price}</span>
          </p>
          <p>
            <label for="status">Status :</label><span>{item.orderStatus}</span>
          </p>
          <p>
            <label for="shippingaddress">Shipping Address :</label><span>{item.orderAddress}</span>
          </p>
        </div>
        <div class="order-commands">
          <p>
            <input class="button" type="button" name="" value="Accept" />
          </p>
          <p>
            <input class="button" type="button" name="" value="Decline" />
          </p>
          <p>
            <input class="button" type="button" name="" value="Mark as Fulfilled" />
          </p>
        </div>
      </div>
    ))
  }
  </div>
);
export default OrderCards;
