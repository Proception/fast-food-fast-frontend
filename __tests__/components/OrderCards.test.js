import React from 'react';
import { shallow } from 'enzyme';
import OrderCards from '../../src/components/OrderCards.jsx';
import { props } from '../../__mocks__/orderCardsMock.js';

let wrapper;
beforeEach(() => {
  const { orders } = props;
  wrapper = shallow(<OrderCards orders={orders}/>);
});
describe('OrderCards Component', () => {
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
