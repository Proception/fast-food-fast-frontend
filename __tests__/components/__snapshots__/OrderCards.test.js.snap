// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderCards Component should match snapshot 1`] = `
<div>
  <div
    class="new-order"
    key="1"
  >
    <div
      class="order-details"
    >
      <p>
        <label
          for="orderno"
        >
          Order No. :
        </label>
        <span>
          12
        </span>
      </p>
      <p>
        <label
          for="orderdate"
        >
          Order Date :
        </label>
        <span>
          30 August, 2016
        </span>
      </p>
      <p>
        <label
          for="totalprice"
        >
          Total Amount :
        </label>
        <span>
          ₦
          5000
        </span>
      </p>
      <p>
        <label
          for="status"
        >
          Status :
        </label>
        <span>
          Pending
        </span>
      </p>
      <p>
        <label
          for="shippingaddress"
        >
          Shipping Address :
        </label>
        <span>
          52, Enitan str, Aguda, Surulere, Lagos State
        </span>
      </p>
    </div>
    <div
      class="order-commands"
    >
      <p>
        <input
          class="button"
          name=""
          type="button"
          value="Accept"
        />
      </p>
      <p>
        <input
          class="button"
          name=""
          type="button"
          value="Decline"
        />
      </p>
      <p>
        <input
          class="button"
          name=""
          type="button"
          value="Mark as Fulfilled"
        />
      </p>
    </div>
  </div>
  <div
    class="new-order"
    key="2"
  >
    <div
      class="order-details"
    >
      <p>
        <label
          for="orderno"
        >
          Order No. :
        </label>
        <span>
          54
        </span>
      </p>
      <p>
        <label
          for="orderdate"
        >
          Order Date :
        </label>
        <span>
          Boiled Yam
        </span>
      </p>
      <p>
        <label
          for="totalprice"
        >
          Total Amount :
        </label>
        <span>
          ₦
          1200
        </span>
      </p>
      <p>
        <label
          for="status"
        >
          Status :
        </label>
        <span>
          In Progress
        </span>
      </p>
      <p>
        <label
          for="shippingaddress"
        >
          Shipping Address :
        </label>
        <span>
          52, Enitan str, Aguda, Surulere, Lagos State
        </span>
      </p>
    </div>
    <div
      class="order-commands"
    >
      <p>
        <input
          class="button"
          name=""
          type="button"
          value="Accept"
        />
      </p>
      <p>
        <input
          class="button"
          name=""
          type="button"
          value="Decline"
        />
      </p>
      <p>
        <input
          class="button"
          name=""
          type="button"
          value="Mark as Fulfilled"
        />
      </p>
    </div>
  </div>
  <div
    class="new-order"
    key="3"
  >
    <div
      class="order-details"
    >
      <p>
        <label
          for="orderno"
        >
          Order No. :
        </label>
        <span>
          67
        </span>
      </p>
      <p>
        <label
          for="orderdate"
        >
          Order Date :
        </label>
        <span>
          30 October, 2017
        </span>
      </p>
      <p>
        <label
          for="totalprice"
        >
          Total Amount :
        </label>
        <span>
          ₦
          1500
        </span>
      </p>
      <p>
        <label
          for="status"
        >
          Status :
        </label>
        <span>
          Fulfilled
        </span>
      </p>
      <p>
        <label
          for="shippingaddress"
        >
          Shipping Address :
        </label>
        <span>
          52, Enitan str, Aguda, Surulere, Lagos State
        </span>
      </p>
    </div>
    <div
      class="order-commands"
    >
      <p>
        <input
          class="button"
          name=""
          type="button"
          value="Accept"
        />
      </p>
      <p>
        <input
          class="button"
          name=""
          type="button"
          value="Decline"
        />
      </p>
      <p>
        <input
          class="button"
          name=""
          type="button"
          value="Mark as Fulfilled"
        />
      </p>
    </div>
  </div>
  <div
    class="new-order"
    key="4"
  >
    <div
      class="order-details"
    >
      <p>
        <label
          for="orderno"
        >
          Order No. :
        </label>
        <span>
          33
        </span>
      </p>
      <p>
        <label
          for="orderdate"
        >
          Order Date :
        </label>
        <span>
          30 January, 2018
        </span>
      </p>
      <p>
        <label
          for="totalprice"
        >
          Total Amount :
        </label>
        <span>
          ₦
          3000
        </span>
      </p>
      <p>
        <label
          for="status"
        >
          Status :
        </label>
        <span>
          Completed
        </span>
      </p>
      <p>
        <label
          for="shippingaddress"
        >
          Shipping Address :
        </label>
        <span>
          52, Enitan str, Aguda, Surulere, Lagos State
        </span>
      </p>
    </div>
    <div
      class="order-commands"
    >
      <p>
        <input
          class="button"
          name=""
          type="button"
          value="Accept"
        />
      </p>
      <p>
        <input
          class="button"
          name=""
          type="button"
          value="Decline"
        />
      </p>
      <p>
        <input
          class="button"
          name=""
          type="button"
          value="Mark as Fulfilled"
        />
      </p>
    </div>
  </div>
</div>
`;
